src/main/java/fi/okm/jod/yksilo/externalapi/v1/ExternalApiV1Service.java

```diff
@@ -26,33 +28,46 @@
 
 /** Bisneslogiikka External API V1 */
 @Service
-@Transactional(readOnly = true)
 @RequiredArgsConstructor
+@Transactional(readOnly = true)
 public class ExternalApiV1Service {
   private final TyomahdollisuusRepository tyomahdollisuusRepository;
   private final KoulutusmahdollisuusRepository koulutusmahdollisuusRepository;
+  private final YksiloRepository yksiloRepository;
 
+  @Transactional
```

Comment:
This annotation overrides the class-level annotation (`@Transactional(readOnly = true)`) making the transaction read-write. You should remove the redundant method-level annotations.